AsyncNats/NatsConnection.cs

```diff
@@ -420,6 +452,58 @@ public async ValueTask PublishMemoryAsync(string subject, ReadOnlyMemory<byte> p
             await WriteAsync(pub, cancellationToken);
         }
 
+
+        //proposed unified apis
+
+        public async ValueTask PublishAsync(NatsKey subject, CancellationToken cancellationToken = default)
+        {        
+            var pub = NatsPub.RentedSerialize(_memoryPool, subject, NatsKey.Empty, NatsPayload.Empty);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+        public async ValueTask PublishAsync(NatsKey subject, NatsMsgHeaders headers, CancellationToken cancellationToken = default)
+        {
+            var pub = NatsHPub.RentedSerialize(_memoryPool, subject, NatsKey.Empty, headers, NatsPayload.Empty);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+        public async ValueTask PublishAsync(NatsKey subject, NatsPayload payload, CancellationToken cancellationToken = default)
+        {
+            var pub = NatsPub.RentedSerialize(_memoryPool, subject, NatsKey.Empty, payload);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+        public async ValueTask PublishAsync(NatsKey subject, NatsMsgHeaders headers, NatsPayload payload, CancellationToken cancellationToken = default)
+        {
+            var pub = NatsHPub.RentedSerialize(_memoryPool, subject, NatsKey.Empty, headers, payload);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+        public async ValueTask PublishAsync(NatsKey subject, NatsKey replyTo, NatsPayload payload, CancellationToken cancellationToken = default)
+        {
+            var pub = NatsPub.RentedSerialize(_memoryPool, subject, replyTo, payload);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+        public async ValueTask PublishAsync(NatsKey subject, NatsKey replyTo, NatsMsgHeaders headers, NatsPayload payload, CancellationToken cancellationToken = default)
+        {
+            var pub = NatsHPub.RentedSerialize(_memoryPool, subject, replyTo, headers, payload);
+            await WriteAsync(pub, cancellationToken);
+        }
+
+
+
+
+#if !DISABLE_PUBLISH_RAW
+        public async ValueTask PublishRaw(ReadOnlyMemory<byte> rawData,int messageCount, CancellationToken cancellationToken)
```

Comment:
Not a fan of this PublishRaw, but I guess it can't really break anything.

Reply:
I'm not a fan either, I just needed a way to send a precomputed batch of messages to enable the benchmark. 
It can probably be compiled with the disable directive when publishing to Nuget.